core/src/main/java/org/sonarsource/sonarlint/core/embedded/server/ShowHotspotRequestHandler.java

```diff
@@ -46,24 +47,27 @@
 import org.sonarsource.sonarlint.core.clientapi.client.message.MessageType;
 import org.sonarsource.sonarlint.core.clientapi.client.message.ShowMessageParams;
 import org.sonarsource.sonarlint.core.commons.TextRange;
+import org.sonarsource.sonarlint.core.repository.connection.ConnectionConfigurationRepository;
 import org.sonarsource.sonarlint.core.serverapi.hotspot.ServerHotspotDetails;
 import org.sonarsource.sonarlint.core.telemetry.TelemetryServiceImpl;
 
 import static org.apache.commons.lang3.StringUtils.isNotBlank;
 
-class ShowHotspotRequestHandler implements HttpRequestHandler {
+@Named
+@Singleton
```

Comment:
If we are using Spring, why not use `@Component` instead of `@Named` and `@Singleton` from Java CDI?

Reply:
My idea is to limit Spring coupling as much as possible, in the (very unlikely, but who knows) case we would like to move to another CDI framework.